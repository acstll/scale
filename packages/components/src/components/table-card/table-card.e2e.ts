import { newE2EPage } from '@stencil/core/testing';

describe('t-table-card', () => {
  it('renders', async () => {
    const page = await newE2EPage();
    await page.setContent('<t-table-card>this is a tag</t-table-card>');
    const element = await page.find('t-table-card');
    expect(element).toHaveClass('hydrated');
  });
});
