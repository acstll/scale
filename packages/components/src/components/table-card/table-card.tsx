import { Component, h, Prop } from '@stencil/core';
import { CssClassMap } from '../../utils/utils';
import classNames from 'classnames';
import { TableRow } from './../table-row/table-row';

@Component({
  tag: 't-table-card',
  styleUrl: 'table-card.css',
  shadow: true,
})
export class TableCard {
  /** (optional) Tag class */
  @Prop() public customClass?: string = '';
  /** (optional) Tag theme */
  @Prop() public theme?: string = '';
  @Prop() public items: string[] = [];

  public render() {
    return (
      <table class={this.getCssClassMap()}>
        {this.items.map(item => {
          return <TableRow item={item} />;
        })}
      </table>
    );
  }

  private getCssClassMap(): CssClassMap {
    return classNames(
      'table-card',
      this.customClass && this.customClass,
      this.theme && `table-card--theme-${this.theme}`
    );
  }
}
