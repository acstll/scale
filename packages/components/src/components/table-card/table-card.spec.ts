import { newSpecPage } from '@stencil/core/testing';
import { TableCard } from './table-card';

describe('TableCard', () => {
  let element;
  beforeEach(async () => {
    element = new TableCard();
  });

  it('should match snapshot', async () => {
    const page = await newSpecPage({
      components: [TableCard],
      html: `<t-table-card>Label</t-table-card>`,
    });
    expect(page.root).toMatchSnapshot();
  });

  it('should handle a custom css class', () => {
    element.customClass = 'custom';
    expect(element.getCssClassMap()).toContain('custom');
  });

  it('should handle theme css class', () => {
    element.theme = 'default';
    expect(element.getCssClassMap()).toContain('table-card--theme-default');
  });

  it('should have a default css class', () => {
    expect(element.getCssClassMap()).toContain('table-card');
  });
});
