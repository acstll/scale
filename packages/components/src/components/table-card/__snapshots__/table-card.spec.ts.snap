// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableCard should match snapshot 1`] = `
<t-table-card>
  <mock:shadow-root>
    <table class="table-card"></table>
  </mock:shadow-root>
  Label
</t-table-card>
`;
